### PR TITLE
Issue #2503205 by Pol, Fabianx: Make it simpler to register CToolsPlugins automatically

### PR DESCRIPTION
**Pull request by fabianx-ai**

### CODE REVIEW

File: lib/Drupal/service_container/Tests/ServiceContainerCToolsIntegrationTest.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    try {
+      $service = \Drupal::service('service_container_test_ctools.yolo')
+        ->createInstance('yolo1');
+    }
+    catch (\Exception $e) {
+      $this->pass('Non-existant plugin does not exist in the container.');
+    }
+
+    // @todo Test and fix instantiating non-class plugins.
```
Comment (fabianx-ai) [inline lib/Drupal/service_container/Tests/ServiceContainerCToolsIntegrationTest.php:70]: @drupol This is not to be fixed here.

File: src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php [hunk 2]
```diff
@@ -221,6 +230,43 @@ public function alterContainerDefinition(&$container_definition) {
   }
 
   /**
+   * Automatically register all ctools plugins of the given types.
+   *
+   * @param @todo
+   * @param @todo
```
Comment (fabianx-ai) [inline src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php:236]: @drupal Please fix this after I merged this PR.

File: src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php [hunk 3]
```diff
@@ -238,6 +284,7 @@ public function processPluginManagers(&$container_definition, $discovery_class,
       }
 
       $container_definition['services'][$name] = $this->getPluginManagerDefinition($name, $discovery_class, $plugin_manager);
+
```
Comment (fabianx-ai) [inline src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php]: self-nit: Unrelated change.

File: tests/modules/service_container_test_ctools/src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * Overrides some specific services/parameters for tests purposes.
+ */
+class ServiceContainerServiceProvider implements ServiceProviderInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getContainerDefinition() {
+    $services = array();
+    $parameters['ctools_plugins_auto_discovery']['service_container_test_ctools'] = TRUE;
```
Comment (fabianx-ai) [inline tests/modules/service_container_test_ctools/src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php:24]: This is the final syntax now :).

Comment (fabianx-ai) [inline tests/modules/service_container_test_ctools/src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php:24]: @drupol ^^

Comment (drupol) [inline tests/modules/service_container_test_ctools/src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php:24]: Excellent :) Thanks !!!

**Comment by fabianx-ai** [comment]
@todo @drupol Also need to update README.md for that.
